src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -40,28 +39,36 @@ class ReturnValue(val expression: String?, val type: Ty) {
     }
 }
 
-class Parameter(val name: String, val type: Ty? = null, val mutability: Mutability = Mutability.IMMUTABLE) {
+class Parameter(
+    val name: String,
+    val type: Ty? = null,
+    val reference: Reference = Reference.NONE
+) {
+
+    enum class Reference(val text: String) {
+        MUTABLE("&mut "), BORROWING("& "), NONE("")
```

Comment:
What about passing a variable by mutable value?
```rust
fn foo(mut v: Vec<i32>) {}

fn main() {
    let mut v = Vec::new();
    <selection>v.push(1);
    foo(v);</selection>
}
```
should be
```rust
fn foo(mut v: Vec<i32>) {}

fn main() {
    let mut v = Vec::new();
    bar(v);
}

fn bar(mut v: Vec<i32>) {
    v.push(1);
    foo(v);
}

```
but current version generates the following
```rust
fn foo(mut v: Vec<i32>) {}

fn main() {
    let mut v = Vec::new();
    bar(&mut v);
}

fn bar(v: &mut Vec<i32>) {
    v.push(1);
    foo(v);
}

```